pages/api/v4/[network]/metadata/token.js

```diff
@@ -92,25 +122,28 @@ const api = async (req, res) => {
 
     let metadata = [];
     if (tokens.length) {
-      metadata = [
-        ...metadata,
-        ...(method === "opensea"
-          ? await Promise.all(
-              await opensea
-                .fetchTokens(chainId, tokens)
-                .then((l) =>
-                  l.map((metadata) => extendMetadata(chainId, metadata))
-                )
-            )
-          : await Promise.all(
-              await rarible
-                .fetchTokens(chainId, tokens)
-                .then((l) =>
-                  l.map((metadata) => extendMetadata(chainId, metadata))
-                )
-            )),
-      ];
+      let newMetadata = null;
+
+      if (method === "opensea") {
+        newMetadata = await opensea
+          .fetchTokens(chainId, tokens)
+          .then((l) => l.map((metadata) => extendMetadata(chainId, metadata)));
+      } else if (method === "rarible") {
+        newMetadata = await rarible
+          .fetchTokens(chainId, tokens)
+          .then((l) => l.map((metadata) => extendMetadata(chainId, metadata)));
+      } else if (method === "simplehash") {
+        newMetadata = await simplehash
+          .fetchTokens(chainId, tokens)
+          .then((l) => {
+            l.map((metadata) => extendMetadata(chainId, metadata));
+            return l;
```

Comment:
I don't think this will work since `map` will not mutate the array, but create a copy of `l`, so we need to return the same as above in order for the map operation to work: `l.map(metadata => extendMetadata(chainId, metadata)`.